packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -71,10 +73,16 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
         if (resource.saveContents) {
             const content = await this.readContents();
             const formattingOptions = { tabSize: 3, insertSpaces: true, eol: '' };
-            const edits = jsoncparser.modify(content, [key], value, { formattingOptions });
-            const result = jsoncparser.applyEdits(content, edits);
-
-            await resource.saveContents(result);
+            try {
```

Comment:
1. why not catching at https://github.com/theia-ide/theia/blob/master/packages/preferences/src/browser/preferences-tree-widget.ts#L138 ?
2. assignment and event emitting happen still if an error occurs, no?

Reply:
>why not catching at https://github.com/theia-ide/theia/blob/master/packages/preferences/src/browser/preferences-tree-widget.ts#L138 ?

because `preferenceService.set()` is called in 4 different places (for now), having the try-catch here would make the messaging work for all.

>assignment and event emitting happen still if an error occurs, no?

Yes, and you are right - I should not let the preferences in memory get updated or event fired. I will make the change. Thank you !

Reply:
@AlexTugarev your feedback has been resolved. could you please check again ?